includes/payment-methods/class-upe-payment-method.php

```diff
@@ -171,11 +171,26 @@ public function is_enabled_at_checkout( string $account_country ) {
 
 		// This part ensures that when payment limits for the currency declared, those will be respected (e.g. BNPLs).
 		if ( [] !== $this->limits_per_currency ) {
-			$currency = get_woocommerce_currency();
+			$order = null;
+			if ( is_wc_endpoint_url( 'order-pay' ) ) {
+				$order = wc_get_order( absint( get_query_var( 'order-pay' ) ) );
+				$order = is_a( $order, 'WC_Order' ) ? $order : null;
+			}
+
+			if ( $order ) {
+				$currency = $order->get_currency();
+			} else {
+				$currency = get_woocommerce_currency();
+			}
 			// If the currency limits are not defined, we allow the PM for now (gateway has similar validation for limits).
 			// Additionally, we don't engage with limits verification in no-checkout context (cart is not available or empty).
```

Comment:
Should we be concerned with (or change) this comment, since maybe the pay-for-order could be considered a "no-checkout context"? 👀 

I wonder if, instead, we should perform a check like this:
```php
			$currency = get_woocommerce_currency();
			if ( $order ) {
				$currency = $order->get_currency();
			}

			// If the currency limits are not defined, we allow the PM for now (gateway has similar validation for limits).
			$total = null;
			if ( $order ) {
				$total = $order->get_total();
			} elseif ( isset( WC()->cart ) ) {
				$total = WC()->cart->get_total( '' );
			}

			if ( isset( $this->limits_per_currency[ $currency ] ) && ! empty( $total ) ) {
				$amount = WC_Payments_Utils::prepare_amount( $total, $currency );
```